pecos/xmc/base.py

```diff
@@ -1052,6 +1114,17 @@ def nr_labels(self):
         else:
             return self.model_chain[-1].nr_labels
 
+    @property
+    def weight_matrix_type(self):
+        """The type of the weight matrix"""
+        if self.is_predict_only:
+            weight_matrix_value = clib.xlinear_get_int_attr(self.model_chain, "weight_matrix_type")
+            return list(XLINEAR_INFERENCE_MODEL_TYPES.keys())[
+                list(XLINEAR_INFERENCE_MODEL_TYPES.values()).index(weight_matrix_value)
+            ]
+        else:
+            return type(self.model_chain[0].W)
```

Comment:
This function will return a string (one of `["CSC", "HASH_CHUNKED", "BINARY_SEARCH_CHUNKED"]`) when `self.is_predict_only=True` but `scipy.sparse.csc_matrix` when `self.is_predict_only=False`. Also, it seems you are returning the top `W` type in the second case but bottom `W` type in the first case. My suggestion is to return `"CSC"` when `self.is_predict_only=False`.

Reply:
Also, please add docstring about the output of this function.

Reply:
Output for is_predict_only=False has been updated to default to "CSC" and the change has been reflected in the updated docstring.